### PR TITLE
925 evss to lighthouse

### PR DESCRIPTION
**Pull request by anitadewitt-va**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary
- The Virtual Agent Chatbot team is now calling Lighthouse's Benefits Claims service in order to surface Veterans their three most recent open compensation claims instead of the EVSS service as it is sunsetting in August. 
- A majority or the business logic for this controller and tests are duplicated as we have a feature toggle to switch between calling the lighthouse and evss services, but this toggle will be removed once we have successfully integrated with lighthouse. 

## Related issue(s)
- department-of-veterans-affairs/va-virtual-agent#925

## Testing done
- There are unit tests for the specific cases for types of claims (i.e. closed compensation claims, non compensation claims, no compensation claims, etc.) As part of this there were four new cassettes created to mimic this behaviour.

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
- These changes will only impact the Virtual Agent Chatbot on va.gov.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ x  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by nichelous-herndon — APPROVED** [review]

File: app/controllers/v0/virtual_agent/virtual_agent_claim_controller.rb [hunk 1]
```diff
@@ -32,19 +56,27 @@ def index
         end
       end
 
-      def show
-        claim = EVSSClaim.for_user(current_user).find_by(evss_id: params[:id])
-
-        claim, synchronized = service.update_from_remote(claim)
+      def poll_claims_from_lighthouse
+        lighthouse_client_id = Settings.lighthouse.benefits_claims.access_token.client_id
```
Comment (jerekshoe) [inline app/controllers/v0/virtual_agent/virtual_agent_claim_controller.rb:60]: You don't need to pass the client_id or rsa_key_path if you are using the ones from `Settings.lighthouse.benefits_claims.access_token`. The service will use these by default if neither is provided

File: app/controllers/v0/virtual_agent/virtual_agent_claim_controller.rb [hunk 2]
```diff
@@ -97,6 +129,54 @@ def get_updated_date(claim)
...
+        return [] if comp_claims.nil?
+
+        transform_claims_to_response_lighthouse(comp_claims)
+      end
+
+      def three_most_recent_open_comp_claims_lighthouse(claims)
+        claims['data']
+          .sort_by { |claim| transform_to_date(claim['attributes']['claimPhaseDates']['phaseChangeDate']) }
+          .reverse
+          .select { |claim| open_compensation_lighthouse? claim }
```
Comment (jerekshoe) [inline app/controllers/v0/virtual_agent/virtual_agent_claim_controller.rb:144]: I would do the `select` before the `sort_by` as that will make reduce the amount of time the `sort_by` will take to complete

File: app/controllers/v0/virtual_agent/virtual_agent_claim_controller.rb [hunk 3]
```diff
@@ -11,7 +12,30 @@ class VirtualAgentClaimController < ApplicationController
...
       def index
+        if Flipper.enabled?(:virtual_agent_lighthouse_claims, @current_user)
+          poll_claims_from_lighthouse
+        else
+          poll_claims_from_evss
+        end
+      end
+
+      def show
+        claim = EVSSClaim.for_user(current_user).find_by(evss_id: params[:id])
```
[the remark this answers is not in the document]
Comment (anitadewitt-va) [reply app/controllers/v0/virtual_agent/virtual_agent_claim_controller.rb:26]: We no longer need this data so this method will be deprecated after EVSS sunsets.

File: app/controllers/v0/virtual_agent/virtual_agent_claim_controller.rb (hunk 1 above)
Comment (nichelous-herndon) [reply app/controllers/v0/virtual_agent/virtual_agent_claim_controller.rb]: We addressed your concern. Please confirm.

File: app/controllers/v0/virtual_agent/virtual_agent_claim_controller.rb (hunk 2 above)
Comment (nichelous-herndon) [reply app/controllers/v0/virtual_agent/virtual_agent_claim_controller.rb]: We addressed your concern. Please confirm.

File: app/controllers/v0/virtual_agent/virtual_agent_claim_controller.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (nichelous-herndon) [reply app/controllers/v0/virtual_agent/virtual_agent_claim_controller.rb]: We addressed your concern. Please confirm.

**Review by jerekshoe — APPROVED** [review]

**Review by holdenhinkle — APPROVED** [review]
Looks good, Aaron!

**Pull request merged**